src/lib/verifier/DAVerifier.sol

```diff
@@ -136,6 +140,34 @@ library DAVerifier {
         return true;
     }
 
+    /// @notice computes the Celestia block square size from a row/column root to data root binary merkle proof.
+    /// Note: the provided proof is not authenticated to the QGB smart contract. It is the user's responsibility
+    /// to verify that the proof is valid and was successfully committed.
+    /// Note: the minimum square size is 1. Thus, we don't expect the proof to have number of leaves equal to 0.
```

Comment:
[question for reviewers]
Does this make sense to be added as a check? I am not sure but this case will never be hit and if we add it as a check it will only waste gas.
What do you think?

Reply:
hmm :thinking: how does relate to the three possibilities for proofs that we discussed before? is this needed to know if some claimed location is outsides of the bounds of the square?

Reply:
> is this needed to know if some claimed location is outsides of the bounds of the square?

Yes. We need to compute the square size.

> how`` does relate to the three possibilities for proofs that we discussed before?

The question itself is not related to the proofs, it's just asking for your opinion for the gas optimization done, that's all

Reply:
I guess what I'm trying to ask is if this check is important to that in any way. 

Like, given a proof we should either return 1) true: the proof is valid 2) false: the proof is invalid or 3) error: the proof is outside the bounds  of the square

this looks like its related to if not the third check. If that's then case then I think that answers the question as to if we need this or not

Reply:
The DAVerifier library doesn't know about fraud proofs, it only provides helper functions related to inclusion proofs.
So, this library will give you a method to calculate the square size from row proof, and then you can use that value separately to see if the sequence of spans is out of bounds

Reply:
is this check important to returning an error that the proof is out of bounds? if so, then I'm saying that we need to keep it

Reply:
No, in that case the returned square size would be 0 and when they will compare it with the index of the blob, it will be out of bounds

Reply:
I think this check isn't strictly necessary. Would be a good point for auditors to confirm.

Reply:
cool, I'll just leave the comment there until it's audited. Thanks